src/harness/testcases/WINNF_FT_S_EXZ_testcase.py

```diff
@@ -0,0 +1,315 @@
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+from datetime import datetime
+import json
+import logging
+import os
+import jwt
+
+import sas
+import sas_testcase
+from util import winnforum_testcase, configurable_testcase, writeConfig, \
+  loadConfig, generateCpiRsaKeys, generateCpiEcKeys, convertRequestToRequestWithCpiSignature
+
+
+class ExclusionZoneTestcase(sas_testcase.SasTestCase):
+
+  def setUp(self):
+    self._sas, self._sas_admin = sas.GetTestingSas()
+    self._sas_admin.Reset()
+
+  def tearDown(self):
+    pass
+
+  def generate_EXZ_1_default_config(self, filename):
+    """Generates the WinnForum configuration for EXZ.1."""
+    # Load device info
+    # Load Devices located outside 50 meters of exclusion zones
+    device_a = json.load(
+        open(os.path.join('testcases', 'testdata', 'device_a_exz.json')))
+    # Load Devices located within at least one exclusion zones
+    device_b = json.load(
+        open(os.path.join('testcases', 'testdata', 'device_b_exz.json')))
+    # Load Devices located within 50 meters of at least one exclusion zones
+    device_c = json.load(
+        open(os.path.join('testcases', 'testdata', 'device_c_exz.json')))
+   
+    # Prepare 3 grant requests.
+    # 1. With frequency range partially overlapping with the frequency 
+    #     range of the nearest exclusion zone.
+    grant_0 = json.load(
+        open(os.path.join('testcases', 'testdata', 'grant_0.json')))
+    grant_0['operationParam']['operationFrequencyRange'] = {
+        'lowFrequency': 3625000000.0,
+        'highFrequency': 3635000000.0
+    }
+
+    # 2. With frequency range partially overlapping with the frequency 
+    #     range of the nearest exclusion zone.
+    grant_1 = json.load(
+        open(os.path.join('testcases', 'testdata', 'grant_0.json')))
+    grant_1['operationParam']['operationFrequencyRange'][
+        'lowFrequency'] = 3625000000.0
+    grant_1['operationParam']['operationFrequencyRange'][
+        'highFrequency'] = 3635000000.0
+
+    # 3. With frequency range partially overlapping with the frequency 
+    #     range of the nearest exclusion zone.
+    grant_2 = json.load(
+        open(os.path.join('testcases', 'testdata', 'grant_0.json')))
+    grant_2['operationParam']['operationFrequencyRange'][
+        'lowFrequency'] = 3625000000.0
+    grant_2['operationParam']['operationFrequencyRange'][
+        'highFrequency'] = 3635000000.0
+
+    # Create the actual config.
+    devices = [device_a, device_b, device_c]
```

Comment:
You actually need to have three *categories* of device, not three devices. The first category is expected to pass both reg and grant. The second and third categories are expected to pass reg but fail grant with code 400.

Also not that since the expected code is written in the test case, it does not need to be configurable here.

Reply:
Modified the script for 3 different categories of devices.

Reply:
The changes do not match the test spec. Please re-read the spec and ask for clarification if needed.